wpilibj/src/main/java/edu/wpi/first/wpilibj/TimedRobot.java

```diff
@@ -128,6 +128,8 @@ public void startCompetition() {
         break;
       }
 
+      Tracer.startTrace("TimedRobot");
```

Comment:
This won't give anything useful- this loop wakes up for each callback function, not just the main IterativeRobot loop.

Reply:
Yeah you're right, the same `addPeriodic` can run multiple times per `TimedRobot` loop.
We may have to document against `Tracer` calls inside `addPeriodic` callbacks.
The issue occurs with `endCycle` logic, `endCycle` is implicitly called when a top-level trace is ended.
The issue comes from the fact `endCycle` publishes a 0 for every trace that didn't run that cycle, it does this to make traces that are conditionally run to accurately report the time they took every cycle (0ms if they didn't run). If the notifier loop in `TimedRobot#startCompetition` isn't fully traced each periodic callback would start its own cycle if they traced and by proxy setting all the publishers in `loopFunc` to zero. I have to tweak the enabled/disable logic to make the tracer only enabled during the `loopFunc` (this will only apply to the robots thread).

Reply:
Ok so I have 2 solutions to the `AddPeriodic` issue. One solution is to enable the tracer for the thread at the beginning of the robot thread and disable the tracer after the loop, causing all Tracer calls in `AddPeriodic` to no-op. The other solution is to add machinery to swap out the thread local state temporarily, this would be used for `AddPeriodic` to make them practically behave as a separate thread. The second option is nicer on users but depending on implementation could add a lot of complexity to the utility.